File: examples/linreg/example_dwtest.R
library("mmstat4")
fatalities <- ghload("fatalities.rds")
lm <- lm (US~YR, data=fatalities)
summary(lm)
plot(fatalities$YR, fatalities$US)
abline(lm, col="red")
#
library("car")
durbinWatsonTest(lm)
#
library("lmtest")
dwtest(lm)
dwtest(lm, alternative="two.sided")

File: examples/linreg/example_dwtest.py
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.stats.stattools import durbin_watson
fatalities = r["mmstat4::ghload"]("fatalities.rds")
lm = smf.ols('US ~ YR', data=fatalities).fit()
print(lm.summary())
print(durbin_watson(lm.resid))
